crates/compiler/mono/src/ir.rs

```diff
@@ -9752,6 +10050,69 @@ fn from_can_pattern_help<'a>(
             })
         }
 
+        TupleDestructure {
+            whole_var,
+            destructs,
+            ..
+        } => {
+            // sorted fields based on the type
+            let sorted_elems = {
+                let mut layout_env = layout::Env::from_components(
+                    layout_cache,
+                    env.subs,
+                    env.arena,
+                    env.target_info,
+                );
+                crate::layout::sort_tuple_elems(&mut layout_env, *whole_var)
+                    .map_err(RuntimeError::from)?
+            };
+
+            // sorted fields based on the destruct
+            let mut mono_destructs = Vec::with_capacity_in(destructs.len(), env.arena);
+            let mut destructs_by_index = Vec::with_capacity_in(destructs.len(), env.arena);
+            destructs_by_index.extend(destructs.iter().map(Some));
+
+            let mut elem_layouts = Vec::with_capacity_in(sorted_elems.len(), env.arena);
+
+            // next we step through both sequences of elems. The outer loop is the sequence based
+            // on the type, since not all elems need to actually be destructured in the source
+            // language.
+            //
+            // However in mono patterns, we do destruct all patterns (but use Underscore) when
+            // in the source the elem is not matche in the source language.
+
+            for (index, variable, res_layout) in sorted_elems.into_iter() {
+                if index < destructs.len() {
+                    // this elem is destructured by the pattern
+                    mono_destructs.push(from_can_tuple_destruct(
+                        env,
+                        procs,
+                        layout_cache,
+                        &destructs[index].value,
+                        res_layout,
+                        assignments,
+                    )?);
+                } else {
+                    // this elem is not destructured by the pattern
+                    // put in an underscore
```

Comment:
we should discuss on zulip whether we actually want this behavior. the current inference and compilation scheme here is such that "(x, y) = (1, 2, 3)" is admitted. that works well for records where the destructuring is named and hence intentional, but for tuples I suspect it's a smell.

Reply:
Sounds like the consensus on zulip was that we do indeed want to give an error (warning?) when destructuring tuples that are too long, right?

Am I right in assuming that we should accomplish that by (in type checking) force the 'ext' of the tuple type for patterns (at least, when the pattern doesn't have '..' as its last item) be empty?

Reply:
Thoughts on implementing the zulip consensus as a followup PR?

Reply:
Yeah I think `(x, y) = (1, 2, 3)` should be an error 👍

Reply:
(and followup PR seems fine to me!)